### PR TITLE
Add docs variable and fragments support

### PR DESCRIPTION
**Pull request by Justinterest**
### Add docs variable and fragments support
- global variable & version variable using `{{var.xxx}}`
- fragment import using `{{fragments/xxx.md}}`

### Changes
- add remark plugins to support variable and fragment import

### For Writer
- add variables.json in `https://github.com/taichi-dev/taichi/tree/master/docs`
- add fragments folder in `https://github.com/taichi-dev/taichi/tree/master/docs`
- using `{{var.xxx}}`, remember no space in `{{}}` to avoid unexpected error

### CODE REVIEW

File: website/docs/lang/articles/get-started/index.md [hunk 1]
```diff
@@ -5,18 +5,18 @@ slug: /
 
 # Getting Started
 
-Welcome to the Taichi Language documentation!
+Welcome to the {{var.orgName}} Language documentation!
```
Comment (rexwangcc) [inline website/docs/lang/articles/get-started/index.md:8]: Not sure if we want to parameterize this with orgName or product name (i.e. Taichi Language as var.productName). @Vissidarte-Herman Your call 😅

File: website/docs/lang/articles/get-started/index.md [hunk 2]
```diff
@@ -5,18 +5,18 @@ slug: /
...
 
 ## Installation
 
-To get started with the Taichi Language, simply install it with `pip`:
+To get started with the {{var.orgName}} Language, simply install it with `pip`:
```
Comment (rexwangcc) [inline website/docs/lang/articles/get-started/index.md:12]: Same here. Also, i think making changes in docs repo than the main taichi repo won't persist?

File: website/docs/lang/articles/get-started/index.md [hunk 3]
```diff
@@ -118,39 +118,26 @@ ti.init(arch=ti.cpu)
...
-|    Linux     |   OK    |    OK    |     OK     |    N/A    |      OK      |
-|    macOS     |   OK    |   N/A    |    N/A     |    OK     |     N/A      |
-
-(OK: supported; N/A: not available)
-
-With `arch=ti.gpu`, Taichi will first try to run with CUDA. If CUDA is
-not supported on your machine, Taichi will fall back on Metal or OpenGL.
-If no GPU backend (CUDA, Metal, or OpenGL) is supported, Taichi will
-fall back on CPUs.
+{{fragments/support-backend-platform.md}}
```
Comment (rexwangcc) [inline website/docs/lang/articles/get-started/index.md:121]: Cool!

File: website/plugins/remark-plugins/fragments/fragment/test.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+## test header
```
Comment (rexwangcc) [inline website/plugins/remark-plugins/fragments/fragment/test.md:1]: Shall we clean this up?

File: website/plugins/remark-plugins/fragments/test.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (rexwangcc) [inline website/plugins/remark-plugins/fragments/test.js:1]: Shall we clean this up?

File: website/plugins/remark-plugins/fragments/tokenizer.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    const reg = new RegExp(`{{([\\s\\S]+?)}}`, 'g')
+    const match = reg.exec(value)
+    if (match) {
+      const subvalue = match[0]
+      let sub = match[1].trim()
+      if (!sub || !sub.startsWith(prefix + '/')) {
+        return
+      }
+
+      /* istanbul ignore if */
```
Comment (rexwangcc) [inline website/plugins/remark-plugins/fragments/tokenizer.js:20]: Not sure what this means, probably just ignore if?

File: website/plugins/remark-plugins/variables/test.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (rexwangcc) [inline website/plugins/remark-plugins/variables/test.js:1]: Shall we clean this up as well?

File: website/plugins/remark-plugins/variables/tokenizer.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      }
+
+      sub = sub.substring(4)
+
+      const fromData = data[sub]
+
+      const found = fromData
+      const now = eat.now()
+
+      /* istanbul ignore if */
```
Comment (rexwangcc) [inline website/plugins/remark-plugins/variables/tokenizer.js:26]: Same question :p

**Review by rexwangcc — COMMENTED** [review]
This is pretty dope, thanks for the work! I left my 2 cents.

File: website/docs/lang/articles/get-started/index.md (hunk 1 above)
Comment (Justinterest) [reply website/docs/lang/articles/get-started/index.md:8]: This file just for test the variable and fragment, you can rollback to original content.

File: website/plugins/remark-plugins/fragments/fragment/test.md (hunk 4 above)
Comment (Justinterest) [reply website/plugins/remark-plugins/fragments/fragment/test.md:1]: This file is test file, just for development to test remark plugin, you can delete as well.

File: website/plugins/remark-plugins/fragments/test.js (hunk 5 above)
Comment (Justinterest) [reply website/plugins/remark-plugins/fragments/test.js:1]: This file is test file, just for development to test remark plugin, you can delete as well.

File: website/plugins/remark-plugins/fragments/tokenizer.js (hunk 6 above)
Comment (Justinterest) [reply website/plugins/remark-plugins/fragments/tokenizer.js:20]: image
on silent mode, just detect node, but not consume

File: website/plugins/remark-plugins/variables/test.js (hunk 7 above)
Comment (Justinterest) [reply website/plugins/remark-plugins/variables/test.js:1]: yes

File: website/docs/lang/articles/get-started/index.md (hunk 1 above)
Comment (Vissidarte-Herman) [reply website/docs/lang/articles/get-started/index.md:8]: LGTM! @jinfenggt Please rollback and create a PR in the https://github.com/taichi-dev/taichi to add the corresponding folders. Cheers!

File: website/plugins/remark-plugins/fragments/fragment/test.md (hunk 4 above)
Comment (k-ye) [reply website/plugins/remark-plugins/fragments/fragment/test.md:1]: If it is not meant to be checked in, just delete it.

**Review by Vissidarte-Herman — APPROVED** [review]
LGTM! Seems all the test codes and test demos have been removed.

File: website/plugins/remark-plugins/fragments/tokenizer.js (hunk 6 above)
Comment (rexwangcc) [reply website/plugins/remark-plugins/fragments/tokenizer.js:20]: I was confused by the word "istanbul", google tells me it's a JS coverage tool: https://github.com/gotwarlost/istanbul, makes sense now.

**Review by rexwangcc — APPROVED** [review]
LGTM!

**Pull request merged**